adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -173,22 +269,38 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
    *
    * @return Returns a new RDD containing sorted data.
    *
-   * @note Does not support data that is unaligned or where objects align to
-   *   multiple positions.
+   * @note Uses ValidationStringency to handle data that is unaligned or where objects
+   *   align to multiple positions.
    * @see sort
    */
-  def sortLexicographically(): U = {
-    replaceRdd(rdd.sortBy(elem => {
+  def sortLexicographically(partitions: Int = rdd.partitions.length,
+                            stringency: ValidationStringency = ValidationStringency.STRICT)(
+                              implicit tTag: ClassTag[T]): U = {
+
+    val partitionedRDD = rdd.keyBy(elem => {
       val coveredRegions = getReferenceRegions(elem)
 
-      require(coveredRegions.nonEmpty, "Cannot sort RDD containing an unmapped element %s.".format(
-        elem))
+      // We don't use ValidationStringency here because multimapped elements
+      // break downstream methods.
       require(coveredRegions.size == 1,
         "Cannot sort RDD containing a multimapped element. %s covers %s.".format(
           elem, coveredRegions.mkString(",")))
 
-      coveredRegions.head
-    }))
+      if (coveredRegions.isEmpty) {
+        throwWarnOrNone[ReferenceRegion](
+          "Cannot sort RDD containing an unmapped element %s.".format(elem),
+          stringency)
+      } else {
+        Some(coveredRegions.head)
+      }
+    }).filter(f => f._1.nonEmpty).map(f => (f._1.get, f._2))
+      .sortBy(elem => elem._1, ascending = true, numPartitions = partitions)
+
+    partitionedRDD.cache()
```

Comment:
I thought we were going to pull the partition mapping out of the sort code, so that people could sort without computing the partition map? Also, storage level should be parametrizable.